assets/src/design-system/components/textArea/index.js

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { forwardRef, useMemo, useRef, useState } from 'react';
+import styled, { css } from 'styled-components';
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+import { Text } from '../typography';
+import { themeHelpers, THEME_CONSTANTS } from '../../theme';
+import { focusCSS } from '../../theme/helpers';
+import labelAccessibilityValidator from '../../utils/labelAccessibilityValidator';
+import useInputEventHandlers from '../../utils/useInputEventHandlers';
+
+const Container = styled.div`
+  position: relative;
+  width: 100%;
+  min-width: 100px;
+`;
+
+const Label = styled(Text)`
+  margin-bottom: 12px;
+`;
+
+const Hint = styled(Text)`
+  margin-top: 12px;
+  color: ${({ hasError, theme }) =>
+    theme.colors.fg[hasError ? 'negative' : 'tertiary']};
+`;
+
+const InputContainer = styled.div(
+  ({ focused, hasError, theme }) => css`
+    box-sizing: border-box;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    padding: 8px 12px;
+    border: 1px solid
+      ${theme.colors.border[hasError ? 'negativeNormal' : 'defaultNormal']};
+    border-radius: ${theme.borders.radius.small};
+    overflow: hidden;
+
+    ${focused &&
+    !hasError &&
+    css`
+      border-color: ${theme.colors.border.defaultActive};
+    `};
+
+    :focus-within {
+      ${focusCSS(theme.colors.border.focus)};
+    }
+  `
+);
+
+const StyledTextArea = styled.textarea(
+  ({ theme }) => css`
+    height: 100%;
+    width: 100%;
+    padding: 0;
+    background-color: inherit;
+    border: none;
+    outline: none;
+    color: ${theme.colors.fg.primary};
+    resize: none;
```

Comment:
💭 Let's add the scrollbar css here so that it's easier to see when we can scroll the hidden content
```suggestion
    resize: none;
    ${themeHelpers.scrollbarCSS};
```
![image](https://user-images.githubusercontent.com/22185279/110528166-13973c80-80d5-11eb-9c87-fb6973b3fb13.png)